sdks/java/harness/src/test/java/org/apache/beam/fn/harness/FnApiDoFnRunnerTest.java

```diff
@@ -90,137 +78,10 @@
 
 /** Tests for {@link FnApiDoFnRunner}. */
 @RunWith(JUnit4.class)
-public class FnApiDoFnRunnerTest {
+public class FnApiDoFnRunnerTest implements Serializable {
 
   public static final String TEST_PTRANSFORM_ID = "pTransformId";
 
-  private static class TestDoFn extends DoFn<String, String> {
-    private static final TupleTag<String> mainOutput = new TupleTag<>("mainOutput");
-    private static final TupleTag<String> additionalOutput = new TupleTag<>("output");
-
-    private enum State {
-      NOT_SET_UP,
-      OUTSIDE_BUNDLE,
-      INSIDE_BUNDLE,
-    }
-
-    private State state = State.NOT_SET_UP;
-
-    private BoundedWindow window;
-
-    @Setup
-    public void setUp() {
-      checkState(State.NOT_SET_UP.equals(state), "Unexpected state: %s", state);
-      state = State.OUTSIDE_BUNDLE;
-    }
-
-    // No testing for TearDown - it's currently not supported by FnHarness.
-
-    @StartBundle
-    public void startBundle() {
-      checkState(State.OUTSIDE_BUNDLE.equals(state), "Unexpected state: %s", state);
-      state = State.INSIDE_BUNDLE;
-    }
-
-    @ProcessElement
-    public void processElement(ProcessContext context, BoundedWindow window) {
-      checkState(State.INSIDE_BUNDLE.equals(state), "Unexpected state: %s", state);
-      context.output("MainOutput" + context.element());
-      context.output(additionalOutput, "AdditionalOutput" + context.element());
-      this.window = window;
-    }
-
-    @FinishBundle
-    public void finishBundle(FinishBundleContext context) {
-      checkState(State.INSIDE_BUNDLE.equals(state), "Unexpected state: %s", state);
-      state = State.OUTSIDE_BUNDLE;
-      if (window != null) {
-        context.output("FinishBundle", window.maxTimestamp(), window);
-        window = null;
-      }
-    }
-  }
-
-  /**
-   * Create a DoFn that has 3 inputs (inputATarget1, inputATarget2, inputBTarget) and 2 outputs
-   * (mainOutput, output). Validate that inputs are fed to the {@link DoFn} and that outputs
-   * are directed to the correct consumers.
-   */
-  @Test
-  public void testCreatingAndProcessingDoFn() throws Exception {
```

Comment:
It will be annoying but please convert this test to the new factory from the old factory